docs/UserGuide.md

```diff
@@ -213,10 +221,27 @@ Format: `find [n/NAME_KEYWORD]... [d/DESCRIPTION_KEYWORD]... [start/START_DATE]
 
 Example: `find n/tp n/CS2103T t/meeting start/2022-03-15 end/2022-03-27 p/low p/medium c/true`
 
+Restrictions:
+
+* `NAME_KEYWORD`s and `DESCRIPTION_KEYWORD`s should only contain alphanumeric characters (letters and numbers), and should not contain any space in the middle.
+
+* `START_DATE` amd `END_DATE` should be valid dates, in the format of `YYYY-MM-DD`. The `START_DATE` should not be later than `END_DATE`, if both of them are specified.
+
+* `PRIORITY` should be `low`, `medium` or `high`. This field is case-insensitive, for example, you may input `LOW` instead of `low`.                                                            
+
+* `TAG` should not contain any space. The search on `TAG` is case-insensitive.
+
+* `COMPLETION_STATUS` should be either `true` or `false`. This field is case-insensitive.
+
 <div markdown="span" class="alert alert-primary">:bulb: **Note:**
 You can also search using multiple descriptors (e.g. `find n/book t/CS2103T`) to narrow down your search.
 </div>
 
+<div markdown="span" class="alert alert-primary">:bulb: **Note:**
+* Keywords are case-insensitive
+* Harmonia does not include partial keyword matches, i.e. `NAME_KEYWORD`s and `DESCRIPTION_KEYWORD`s need to be complete words.
```

Comment:
will it be clearer to say "i.e. Only tasks containing keywords that match `NAME_KEYWORD`s and `DESCRIPTION_KEYWORD`s exactly will be retrieved." (cuz what if the task they key in contains short form alrdy HAHA, will it be confusing for the user because they might mistakenly think that they have to key in the complete form of the word)